libraries/AP_HAL_Linux/RCOutput_Bebop.cpp

```diff
@@ -134,12 +135,11 @@ bool RCOutput_Bebop::_get_info(struct bldc_info *info)
         return false;
     }
     memset(info, 0, sizeof(struct bldc_info));
-    if (!_i2c_sem->take(0)) {
+    if (!_dev->get_semaphore()->take(0)) {
         return false;
     }
-    hal.i2c1->readRegisters(BEBOP_BLDC_I2C_ADDR, BEBOP_BLDC_GET_INFO,
-                            sizeof(struct bldc_info), (uint8_t *)info);
-    _i2c_sem->give();
+    _dev->read_registers(BEBOP_BLDC_GET_INFO, (uint8_t*)info, sizeof(info));
```

Comment:
sizeof(info) is the size of a pointer to info.... you need to keep the sizeof as before